server/arm.py

```
@@ -247,46 +346,68 @@ def parse_arm_ack(frame_info):
-    position_approx = position_raw / 64.0
+    exponent     = payload[1]
+    mantissa     = int.from_bytes(payload[2:4], byteorder="big", signed=True)
+    fraction     = mantissa / (2 ** exponent)
+    full_range   = JOINT_FULL_RANGE.get(joint_name, 360.0)
+    motor_approx = round(fraction * full_range, 2)
 
     arm_debug_log(
         f"pos:{joint_name}",
-        f"[ARM RX] {joint_name} raw={position_raw}, "
+        f"[ARM RX] {joint_name} exp={exponent} mant={mantissa} "
         f"approx={position_approx}, frame={frame_info['frame']}",
```

Ruff (0.15.21)

[error] Undefined name `position_approx`

(F821)